picard/track.py

```diff
@@ -308,48 +311,62 @@ def _customize_metadata(self):
         if config.setting['convert_punctuation']:
             tm.apply_func(asciipunct)
 
+    @staticmethod
+    def _genres_to_metadata(genres, limit=None, minusage=0, filters='', join_with=None):
+        if limit is not None and limit < 1:
+            return []
+
+        # Ignore tags with zero or lower score
+        genres = +genres
+        if not genres:
+            return []
+
+        # Find most common genres
+        most_common_genres = genres.most_common(limit)
+        topcount = most_common_genres[0][1]
+
+        # Filter by name and usage
+        genres_filter = TagGenreFilter(filters)
+        genres_list = []
+        for name, count in most_common_genres:
+            if genres_filter.skip(name):
+                continue
+            percent = 100 * count // topcount
```

Comment:
`TagGenreFilter` already has a `filter` method, which we don't actually use. I think we should have `TagGenreFilter.filter(genres)` which accepts tuples of genre + count and yields only the unfiltered ones so we can do:

```python
for name, count in genres_filter.filter(most_common_genres):
    percent = 100 * count // topcount
    ...
```

What do you think?

Reply:
Good idea, done in 8d87135